cinder-{{cookiecutter.driver_name_lc}}/config.yaml

```diff
@@ -1,5 +1,37 @@
 options:
-  volume-dd-blocksize:
-    default: 1024
-    description: Disk block size for the volume
-    type: int
+{%- if cookiecutter.driver_install_method == 'PPA' %}
+  driver-source:
+    type: string
+    default:
+    description: |
+      Optional configuration to support use of additional sources such as:
+        - ppa:myteam/ppa
+        - cloud:trusty-proposed/kilo
+        - http://my.archive.com/ubuntu main
+      The last option should be used in conjunction with the key configuration
+      option.
+  driver-key:
+    type: string
+    default:
+    description: |
+      Key ID to import to the apt keyring to support use with arbitary source
+      configuration from outside of Launchpad archives or PPA's.
+{%- endif %}
+  volume-backend-name:
+    type: string
+    default:
+    description: |
+      Volume backend name for the backend. The default value is the
+      application name in the Juju model.
+  use-multipath:
+    type: boolean
+    default: True
+    description: |
+      Whether to use a multipath connection for iSCSI or FC in Cinder
+      volume service. Enabling multipath for VMs is managed by the
+      "use-multipath" option in the nova-compute charm.
+  protocol:
+    type: string
+    default:
+    description: |
+      SAN protocol to use. Choose between iscsi or fc.
```

Comment:
This is more of a question, but is there any way in the new scheme to declare this is a mondatory config?

Previously we were able to set `mandatory_config` in the class so the application status was automatically set as "blocked" when those options are not set.

Reply:
AFAIK, no, there isn't any equivalent functionality. It's definitely something we need to add to `charm-ops-openstack`, though, so I'll get to it.

Reply:
Mandatory config has now been added to charm-ops-openstack and this class as well.